modules/caddypki/acmeserver/acmeserver.go

```diff
@@ -175,6 +197,16 @@ func (ash *Handler) Provision(ctx caddy.Context) error {
 
 func (ash Handler) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhttp.Handler) error {
 	if strings.HasPrefix(r.URL.Path, ash.PathPrefix) {
+		acmeCtx := acme.NewContext(
+			r.Context(),
+			ash.acmeDB,
+			ash.acmeClient,
+			ash.acmeLinker,
+			nil,
+		)
+		acmeCtx = authority.NewContext(acmeCtx, ash.acmeAuth)
+		r = r.WithContext(acmeCtx)
```

Comment:
This is ok. You're doing this for every request, and it should be fine. A minor optimization would be to build it once and move it to the server BaseContext, this is what we do. But it will be available everywhere. 

Looking at chi, they have a middleware that we've never used named [ServerBaseContext](https://pkg.go.dev/github.com/go-chi/chi@v4.1.2+incompatible#ServerBaseContext). It allows you to define a base context for a handler, and it adds the context added by `net/http`. I don't think it's going to make an impact on the performance, but you can test and find out.

Reply:
@francislavoie Do you want me to hold up on merging until we make that optimization? Or should we go with this and then optimize later?

Reply:
There's no rush on that one. I think we can merge.